Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Corefinement/Visitor.h

```diff
@@ -1496,22 +1560,36 @@ class Surface_intersection_visitor_for_corefinement{
 
     //2)triangulation of the triangle faces containing intersection point in their interior
     //  and also those with intersection points only on the boundary.
+
+
+    int total_size = 0;
+    for (typename std::map<TriangleMesh*,On_face_map>::iterator
+           it=on_face.begin(); it!=on_face.end(); ++it)
+    {
+      total_size += it->second.size();
+    }
+    user_visitor.start_triangulation(total_size);
```

Comment:
```suggestion
  user_visitor.start_triangulation(doing_autorefinement?on_face[tm1_ptr].size():on_face[tm1_ptr].size()+on_face[tm2_ptr].size());
```

Reply:
It's more compact and I was hoping that it can be optimized out by the compiler but since `operator[]` is not const it is unlikely.